mongomock/helpers.py

```diff
@@ -269,6 +270,11 @@ def patch_datetime_awareness_in_document(value):
     # mixing tz aware and naive.
     # On top of that, MongoDB date precision is up to millisecond, where Python
     # datetime use microsecond, so we must lower the precision to mimic mongo.
+    if isinstance(value, OrderedDict):
```

Comment:
The two `if`s can be united by just initializing by the object's type: `type(obj)(...)`

Reply:
Smart. Thanks!